### ISSUE TITLE
enable additional SP metadata

### ISSUE THREAD

**Issue by bmesuere** [onelogin/ruby-saml#433]
My idp requires me to add additional fields (e.g. Organization, ContactPerson, ...) to my metadata before I can register my SP with them.

Is there a way to add these to the generated metadata?

**Comment by pitbulk** [comment]
Not supported.

You will need to extend the [Metadata](https://github.com/onelogin/ruby-saml/blob/master/lib/onelogin/ruby-saml/metadata.rb) class

**Comment by bmesuere** [comment]
Thanks for your quick reply. It would be easier to extend this if `generate` was split into 2 separate methods: 1 to generate the document and 1 to sign and convert to string. The only way to do it right now is to effectively copy the existing code and add a few tags, which is not very future proof.

**Comment by pitbulk** [comment]
Yes other toolkits (php-saml, python-saml) I has that splited.

Right now I see 2 alternatives:
- Extend generate method to accept contact and organization objects that will include info on the metadata.
- Extract the sign on a new static method and add a flag to sign or not the metadata calling the method we previously created, add another method that add contact and organization to the metadata.

**Comment by luax** [comment]
I too would like to add `Organization` and `ContactPerson` to metadata.

> Extend generate method to accept contact and organization objects that will include info on the metadata.

I haven't looked at the code closely but how about adding this data to settings? A quick solution would maybe be an optional `additional_elements` list containing any `REXML::Element` one would want, in this case `md:Organization` and `md:ContactPerson`, for example:
```
settings = OneLogin::RubySaml::Settings.new
org_element = REXML::Element.new("md:Organization")
org_name = org_element.add_element("md:OrganizationName")
settings.additional_elements = [ org_element ]
```

Then in `Metadata.generate`
```
if (elements = settings.additional_elements)
  elements.each { |element| root.add_element element }
end
```

Or instead of any element, it could be more specific with `settings.organization`:
```
settings = OneLogin::RubySaml::Settings.new
settings.organization = { organization_name: "Foo" }
```

Then in `Metadata.generate`
```
if (org = settings.organization)
  org_element = REXML::Element.new("md:Organization")
  if (name = org[:organization_name])
    org_name = org_element.add_element("md:OrganizationName")
    org_name.text = name 
  end
  if (display_name = org[:organization_display_name])
    org_display_name = org_element.add_element("md:OrganizationDisplayName")
    org_display_name.text = display_name
  end
  if (url = org[:organization_url])
    org_url = org_element.add_element("md:OrganizationURL")
    org_url.text = url 
  end
  root.add_element(org)
end
```
It would be harder do make the latter customized to have the ability to add different attributes etc.

What do you think? And thanks for a good library!

**Comment by johnnyshields** [comment]
I've raised a PR to fix this issue: #602
